scripts/download_openssl_quic.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+# If any commands fail, fail the script immediately.
+set -ex
+
+# Clone the curl repository to the specified directory.
+git clone --depth 1 -b openssl-3.1.4+quic https://github.com/quictls/openssl $1
```

Comment:
So this is replacing all of our openssl interactions with a fork, even for non-http3 builds? I'm not sure about that. Feels like we might need to partition off the http3 build. We should be able to create two library directories to cover that.

Reply:
I changed the scripts to create two independent `INSTALLDIR`s, one is built with OpenSSL and the other one has quictls, ngtcp2 and nghttp3. The harnesses then link to the corresponding libcurl.a (all link to the OpenSSL one, except the HTTP3 harness)